lib/ssh_subsystem_fwup.ex

```diff
@@ -198,4 +203,22 @@ defmodule SSHSubsystemFwup do
   def code_change(_old, state, _extra) do
     {:ok, state}
   end
+
+  defp check_devpath(devpath) do
+    if is_binary(devpath) and File.exists?(devpath) do
+      :ok
+    else
+      {:error, "Invalid device path: #{inspect(devpath)}"}
+    end
+  end
+
+  defp precheck(nil, options), do: {:ok, options}
+
+  defp precheck({m, f, a}, options) do
+    case apply(m, f, a) do
+      {:ok, new_options} -> {:ok, Keyword.merge(options, new_options)}
+      {:error, reason} -> {:error, reason}
+      _ -> {:error, "precheck failed"}
```

Comment:
```suggestion
      e -> {:error, "precheck failed for unknown reason - #{inspect(e)}"}
```

Is it worth trying to pass a little more info about an unknow failure?

Reply:
Good point. Adding that line is probably the first thing anyone will do who gets that message.